### PR TITLE
Added docs for AppDelegate.OnCommissioningSessionEstablishmentStarted

### PR DESCRIPTION
**Pull request by sharadb-amazon**
Added documentation to address feedback in https://github.com/project-chip/connectedhomeip/pull/28138#pullrequestreview-1541858963

### CODE REVIEW

File: src/app/server/AppDelegate.h [hunk 1]
```diff
@@ -29,7 +29,7 @@ class AppDelegate
 public:
     virtual ~AppDelegate() {}
     /**
-     * This is called on start of session establishment process
+     * This is called on start of session establishment process i.e. when the PBKDFParamRequest is received
      */
```
Comment (cliffamzn) [inline src/app/server/AppDelegate.h:33]: in the interest of making the code base better, can't we just add docs to the other functions as well? The documentation becomes more useful in the more shared APIs like this one.

File: src/app/server/AppDelegate.h [hunk 2]
```diff
@@ -29,7 +29,7 @@ class AppDelegate
 public:
     virtual ~AppDelegate() {}
     /**
-     * This is called on start of session establishment process
+     * This is called on start of session establishment process i.e. when the PBKDFParamRequest is received
```
Comment (cliffamzn) [inline src/app/server/AppDelegate.h:32]: Should we just make it more explicit? Maybe something like this?

```
     * This is called when the PBKDFParamRequest is received and indicates the session establishment process ```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/server/AppDelegate.h (hunk 1 above)
Comment (sharadb-amazon) [reply src/app/server/AppDelegate.h:33]: Addressing this and the comment below in this new PR: #28138

**Comment by sharadb-amazon** [comment]
Adding these docs and more in this new PR: [#28138](https://github.com/project-chip/connectedhomeip/pull/28138)

**Pull request closed without merging**